### PR TITLE
plutus-tx: convert dependent functions to use unfoldings rather than TH

### PR DESCRIPTION
**Pull request by michaelpj**
Seems to be going okay so far, but I had to mess with the playground a bit.

I'll move on to `plutus-wallet-api` next.

### CODE REVIEW

**Comment by michaelpj** [comment]
Eek, that doesn't happen locally, that shouldn't be possible :scream:

**Comment by michaelpj** [comment]
Ugh, it's only on darwin, of course.

**Comment by michaelpj** [comment]
That was an enormous pain. I had to come up with two different workarounds for Haddock and doctests, but I have managed to get both of them working, and I think the workarounds are relatively stable.

File: nix/overlays/required.nix [hunk 1]
```diff
@@ -7,11 +7,14 @@ let
   addRealTimeTestLogs = drv: overrideCabal drv (attrs: {
     testTarget = "--show-details=streaming";
   });
+  -- We do this for things where we need to run the plugin, but where Haddock chokes on it. We can't just turn it off,
```
Comment (cleverca22) [inline nix/overlays/required.nix]: nix uses `#` for comments, this breaks CI

Comment (michaelpj) [reply nix/overlays/required.nix]: :man_facepalming: This does actually work, I just added the comment afterwards.

**Comment by michaelpj** [comment]
Sorry, keep changing my mind here. I think we should merge this soonish as it has the build-related changes. I'll keep doing further changes to the object-level code in follow up PRs.

Will merge in a bit pending comments.

**Pull request merged**